src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,121 @@
+            String[] taskInfo = tempInfo.split(" \\| ");
+            String taskType = taskInfo[0];
+            boolean isDone = taskInfo[1].equals("1");
+            String des = taskInfo[2];
+
+            Task task;
+            switch (taskType) {
+            case "T":
+                task = new ToDo(des);
+                break;
+            case "D":
+                if (taskInfo.length > 3) {
+                    task = new Deadline(des, taskInfo[3]);
+                } else {
+                    System.out.println("Hey man, I am missing info for deadline task in storage file.");
+                    continue;
+                }
+                break;
+            case "E":
+                if (taskInfo.length > 4) {
+                    task = new Event(des, taskInfo[3], taskInfo[4]);
+                } else {
+                    System.out.println("Hey man, I am missing info for event task data in storage file.");
+                    continue;
+                }
+                break;
+            default:
+                System.out.println("Well, I have never seen this task type before.");
+                continue;
+            }
+
+            if (isDone) {
+                task.markAsDone();
+            }
+            tasks.add(task);
+        }
+        reader.close();
+        return tasks;
+    }
+
+    public void save() throws IOException {
+        save(null);
+    }
+
+    public void save(List<Task> tasks) throws IOException {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(filePath));
+        for (Task task : tasks) {
+            writer.write(taskToFile(task));
+            writer.newLine();
+        }
+        writer.close();
+    }
+
+    /**
+     * Writes the tasks onto the file.
+     *
+     * @param task which has been written
+     * @return task onto the handsome.txt file
+     */
+    private String taskToFile(Task task) {
+        StringBuilder storedTask = new StringBuilder();
+
+        if (task instanceof ToDo) {
+            storedTask.append("T");
+            storedTask.append(" | ").append(task.isDone ? "1" : "0");
+            storedTask.append(" | ").append(task.description);
+        } else if (task instanceof Deadline) {
+            storedTask.append("D");
+            storedTask.append(" | ").append(task.isDone ? "1" : "0");
+            storedTask.append(" | ").append(task.description);
+            storedTask.append(" | ").append(DESIRED_FORMAT.format(((Deadline) task).by));
+        } else if (task instanceof Event) {
+            storedTask.append("E");
+            storedTask.append(" | ").append(task.isDone ? "1" : "0");
+            storedTask.append(" | ").append(task.description);
+            storedTask.append(" | ").append(DESIRED_FORMAT.format(((Event) task).from));
+            storedTask.append(" | ").append(DESIRED_FORMAT.format(((Event) task).to));
+        }
+        return storedTask.toString();
+    }
```

Comment:
Consider creating an abstract ```toFileString()``` in your Task class and override the method in your ToDo, Deadline and Event classes depending on the format required. Would help with repeated code and maintaining your code if you add new types of Tasks in the future.